File: 0995-minimum-number-of-k-consecutive-bit-flips/0995-minimum-number-of-k-consecutive-bit-flips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        n, fliped, ans = len(nums), 0, 0
        for i in range(n):
            if i >= k and nums[i-k] > 1:
                fliped -= 1
            if fliped % 2 == nums[i]:
                if i + k > n:
                    return -1
                nums[i] = 2
                fliped += 1
                ans += 1
        return ans

File: 0995-minimum-number-of-k-consecutive-bit-flips/0995-minimum-number-of-k-consecutive-bit-flips.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minKBitFlips = function(nums, k) {
  const n = nums.length; 
  let [fliped, ans] = [0, 0];
  
  for (let i = 0; i < n; i++) {
    if (i >= k && nums[i - k] > 1) fliped--;
    if (fliped % 2 === nums[i]) {
      if (i + k > n) return -1;
      
      nums[i] = 2;
      fliped++;
      ans++;
    }
  }
  
  return ans;
};